tools/docker/dev_container.Dockerfile

```diff
@@ -1,11 +1,9 @@
 #syntax=docker/dockerfile:1.1.5-experimental
-FROM gcr.io/tensorflow-testing/nosla-cuda11.2-cudnn8.1-ubuntu18.04-manylinux2010-multipython as dev_container_cpu
+ARG PY_VERSION
+FROM tensorflow/build:latest-python$PY_VERSION as dev_container_cpu
```

Comment:
If you can add here a fake but future-proof `IMAGE_TYPE` arg I think that we could remove the overhead of maintaining another separated Dockerfile (if you suppose that the our devel and our Vscode/Codespaces image could be almost the same one):
https://github.com/tensorflow/addons/blob/4617bc2e7e2a90b27ed9e070c36fc0724c99e149/.devcontainer/Dockerfile#L1-L2

The main thing we are doing on that additional Dockerfile, other then let the user controlling the `IMAGE_TYPE` when they need the GPU image (currently not available)  is that we are adding a `default` (guid/uid) non root user in the image build.

If we exclude that unification we need to evaluate if without a default non root user, we can still official suggest that we prefer/support only [Docker in rootless mode](https://docs.docker.com/engine/security/rootless/) this is currently the SIG Build policy that doesn't want to add a "default" user in the image.

It is hard to support both as rootless Docker vs root Docker are mutually exclusive for root and non root users permission on the host mounted volumes:

https://github.com/moby/moby/issues/41497
https://discuss.tensorflow.org/t/adopting-open-source-dockerfiles-for-official-tf-nightly-ci/6050/9

Reply:
Added the holder for `IMAGE_TYPE`. And updated to push it correctly as gpu dev container. I think it should be good to merge and we can revisit once we have a means to get a cpu image.